docker/main/ngen/Dockerfile

```diff
@@ -783,7 +788,25 @@ FROM rocky-ngen-deps as rocky_ngen_build_testing
 #FROM rocky_build_ngen as rocky_ngen_build_testing
 #FROM ${DOCKER_INTERNAL_REGISTRY}/ngen-deps:latest as rocky_ngen_build_testing
 
+ARG NUMPY_CONSTRAINT
```

Comment:
This is an abuse of the args parameter. Do not use this to inject python versions. Stick to setup.py and requirements.txt.